### PR TITLE
chore(styles): remove v9 styles

### PR DESCRIPTION
**Pull request by asudoh**
Refs #2308.

#### Changelog

**Removed**

- Deprecated variables, functions and mix-ins that are meant to be removed for `v10`. The notable ones that are kept are:
    - Helvetica font support code
    - Classic grid code
    - `rem()`/`em()` and their dependencies, given references found in `v10` codebase. Will migrate those and remove `rem()`/`em()` in a separate PR
    - `$base-font-size`, as above depends on this
    - Deprecated color tokens
- Tests for above
- `v9` component styles
- Dev env code that are specific to `v9`
- Some classic color references in dev env code, affecting dev env only.

#### Testing / Reviewing

Testing should make sure no component is broken.

### CODE REVIEW

File: src/components/list-box/_list-box.scss [hunk 1]
```diff
@@ -18,284 +18,9 @@
 
 $list-box-width: 100%;
 $list-box-height: rem(40px);
-$list-box-inline-height: if(not feature-flag-enabled('components-x'), rem(32px), $list-box-height);
+$list-box-inline-height: $list-box-height;
```
Comment (emyarod) [inline src/components/list-box/_list-box.scss:21]: I think some of these variables are no longer being used

**Review by emyarod — APPROVED** [review]
looks good to me!

**Comment by mattrosno** [comment]
Hi @asudoh, I've been working on https://github.com/IBM/carbon-components/issues/2266, draft PR https://github.com/IBM/carbon-components/pull/2306. Let me review this prior to merging, to see what needs merged where first, as both PRs touch almost every Sass file.

**Review by jnm2377 — APPROVED** [review]
LGTM! 🎉 thanks for doing this!!

File: demo/scss/_layout.scss [hunk 2]
```diff
@@ -75,14 +70,9 @@ body {
 }
 
 a {
-  @if not feature-flag-enabled('components-x') {
-    @include font-smoothing;
-  }
-  color: $color__blue-51;
   font-weight: 600;
 
   &:hover {
```
Comment (mattrosno) [inline demo/scss/_layout.scss]: Remove this declaration block since it's empty? (The `:hover`.)

File: src/components/accordion/_accordion.scss [hunk 3]
```diff
@@ -8,141 +8,8 @@
 @import '../../globals/scss/vars';
 @import '../../globals/scss/helper-mixins';
 @import '../../globals/scss/layout';
-@import '../../globals/scss/functions';
 @import '../../globals/scss/vendor/@carbon/elements/scss/import-once/import-once';
 @import '../../globals/scss/css--reset';
-@import '../../globals/scss/css--typography';
-
-@mixin accordion {
```
Comment (mattrosno) [inline src/components/accordion/_accordion.scss]: Applies to all components, accordion example here, why not rename `accordion--x()` to `accordion()` after deleting the v9 accordion mixin?

These aren't explicitly marked as "private" like they should be... but in a future PR for inline theming, we'll want one entry point public mixin per component. So what you have is good knowing that we'll be modifying component mixin naming later?

File: src/components/data-table-v2/_data-table-v2-sort.scss [hunk 4]
```diff
@@ -9,64 +9,6 @@
 @import '../../globals/scss/vendor/@carbon/elements/scss/import-once/import-once';
 @import '../../globals/scss/vars';
 
-@mixin data-table-v2-sort {
```
Comment (mattrosno) [inline src/components/data-table-v2/_data-table-v2-sort.scss:12]: Data table Sass lives in `data-table` and `data-table-v2` directories, where one imports the other but yet some files look like duplicates of the other? Can all data table styling live in one directory?

File: src/globals/scss/_theme-tokens.scss [hunk 5]
```diff
@@ -261,27 +130,23 @@ $carbon--theme--v9: (
...
   $active-01: $active-ui !default !global;
-  $nav-01: $color__navy-gray-1 !default !global;
-  $nav-02: $color__blue-90 !default !global;
-  $nav-03: $color__purple-30 !default !global;
-  $nav-04: $color__purple-60 !default !global;
-  $nav-05: $color__teal-40 !default !global;
-  $nav-06: $color__teal-50 !default !global;
-  $nav-07: $color__blue-30 !default !global;
-  $nav-08: $color__blue-51 !default !global;
+  $nav-01: #0f212e !default !global; // Old $color__navy-gray-1
```
Comment (mattrosno) [inline src/globals/scss/_theme-tokens.scss:140]: This PR is to remove v9 styles. Does that also include items that had been deprecated like `$nav-01`? Searching `@deprecated` revealed a number of deprecated items in our source still.

**Review by mattrosno — CHANGES_REQUESTED** [review]

File: src/components/accordion/_accordion.scss (hunk 3 above)
Comment (asudoh) [reply src/components/accordion/_accordion.scss]: Was on the fence on this, but I can definitely do this.

File: demo/scss/_layout.scss (hunk 2 above)
Comment (asudoh) [reply demo/scss/_layout.scss]: Good catch - Will do.

File: src/components/data-table-v2/_data-table-v2-sort.scss (hunk 4 above)
Comment (asudoh) [reply src/components/data-table-v2/_data-table-v2-sort.scss:12]: `_data-table-v2.scss` importing `_data-table.scss` is for keeping backward-compatibility, and need to keep it until `v11`.

File: src/globals/scss/_theme-tokens.scss (hunk 5 above)
Comment (asudoh) [reply src/globals/scss/_theme-tokens.scss:140]: Thank you for bringing this up - I didn't remove as `v10` code still has reference to `$hover-field`, etc. Added this to "non-removed" list in the issue description. Will bring this up as a discussion and will remove it in a separate PR if our consensus is to remove.

File: src/components/accordion/_accordion.scss (hunk 3 above)
Comment (mattrosno) [reply src/components/accordion/_accordion.scss]: I'm cool with keeping `--x()` mixin naming in for now. Even though not explicitly "private", they should be, so we can rename later to not have "--x" suffix when the time is right. Possibly when each component has a public entry point mixin.

File: src/components/data-table-v2/_data-table-v2-sort.scss (hunk 4 above)
Comment (mattrosno) [reply src/components/data-table-v2/_data-table-v2-sort.scss:12]: 2 of the v2 data table Sass files don't import the v1 files. They seem to have duplicate content. Can you verify?

**Review by mattrosno — APPROVED** [review]

**Pull request merged**